woocommerce/woocommerce-gutenberg-products-block
# Fix Twenty Twenty One Price filter, Active filters and radio control styling

Part of #3433.

This PR fixes the Twenty Twenty One styling of:
* Filter products by price block.
* Active filters block.
* Radio control component (used in the shipping options of the Cart and Checkout blocks).

It also refactors the radio control styles so the file `_mixins.scss` is removed. It was there for legacy reasons, but in this PR everything has been moved to a `style.scss`.

### Screenshots

#### Active filters
_Before:_
![imatge](https://user-images.githubusercontent.com/3616980/99972946-7b807c00-2d9f-11eb-8e42-40b1fdaf5809.png)

_After:_
![imatge](https://user-images.githubusercontent.com/3616980/99973330-efbb1f80-2d9f-11eb-87f2-c8e1bed0d5c2.png)

#### Price filter
_Before:_
![imatge](https://user-images.githubusercontent.com/3616980/99973014-981cb400-2d9f-11eb-81b9-f02c30597a67.png)

_After:_
![imatge](https://user-images.githubusercontent.com/3616980/99973288-e336c700-2d9f-11eb-9f82-4d53947e29ea.png)

_After (IE11):_
![imatge](https://user-images.githubusercontent.com/3616980/99974361-19287b00-2da1-11eb-9a5c-30a2e0bd1a1f.png)

:point_up_2: Styling is not perfect in IE11, but everything should work fine. Twenty Twenty One has several other glitches on IE11, so I think it's ok not getting a pixel perfect design in IE11.

#### Radio control (in Checkout block)
_Before:_
![imatge](https://user-images.githubusercontent.com/3616980/99972894-60ae0780-2d9f-11eb-9904-d662b94ed9df.png)

_After:_
![imatge](https://user-images.githubusercontent.com/3616980/99973215-d2865100-2d9f-11eb-90b3-772e5dee7ae1.png)

### How to test the changes in this Pull Request:

1. With the Twenty Twenty One theme installed, create a page with the Filter Products by Price, Active filters and All Products blocks. Create another page with the Checkout block (ensuring you have at least two shipping methods enabled).
2. Both, in the editor and the frontend verify all blocks look correct. (In the screenshots above you can see what has been changed in this PR).
3. It would be good to test that in some other themes (at least, Storefront and Twenty Twenty) that there are no regression.
4. The price slider component is quite complex, so it would be good to test it with different browsers as well.

### Known issues

Filter products by price block styles are not correctly applied in the editor:

![imatge](https://user-images.githubusercontent.com/3616980/99973612-3dd02300-2da0-11eb-9f68-d7139e13c61b.png)

That's because form styles are not applied in the editor in Twenty Twenty One. I opened an issue in its repo (https://github.com/WordPress/twentytwentyone/issues/862) which is now closed but I would like to follow-up later today or tomorrow.

cc @mikejolley
